Dockerfile

```diff
@@ -0,0 +1,38 @@
+# Build stage
+FROM python:3.10-slim AS build
+
+# Set work directory
+WORKDIR /app
+
+# Copy only requirements.txt to install dependencies
+COPY requirements.txt .
+
+# Install dependencies
+RUN pip install --no-cache-dir -r requirements.txt
+
+# Copy application files
+COPY . .
```

Comment:
The application files are copied twice, once in the build stage and again in the run stage. This redundancy should be removed to streamline the Dockerfile. Consider copying only the necessary files in the run stage.